tests/python/sdk/test_client.py

```diff
@@ -160,6 +160,49 @@ def mocked_version(_):
     assert "Server version '0' is not compatible with SDK version" in logger_stream.getvalue()
 
 
+@pytest.mark.parametrize(
+    "server_version, supported_version",
+    [
+        ("3.2", "2"),
+        ("2", "2.1"),
+        ("2.1", "2.1"),
+        ("2.1.1", "2.1"),
+        ("2.2", "2.1"),
+        ("2.2", "2.3"),
+        ("2.1.0.dev123", "2.1.post2"),
+        ("1!1.3", "2.1"),
+        ("1!1.1.dev12", "1!1.1"),
+    ],
+)
+def test_can_check_server_version_compatibility(
+    fxt_logger: Tuple[Logger, io.StringIO],
+    monkeypatch: pytest.MonkeyPatch,
+    server_version: str,
+    supported_version: str,
+):
+    logger, _ = fxt_logger
+
+    supported_v = pv.Version(supported_version)
+    server_v = pv.Version(server_version)
+
+    # Currently, it is ~=, as defined in https://peps.python.org/pep-0440/
+    expected_true = (
```

Comment:
This should be part of the test parameters as a hardcoded value. Otherwise, you're just testing your code against a copy of itself.

Also, it's called `expected_true`, but... true what? You never actually check for a true value in this test.

Reply:
> This should be part of the test parameters as a hardcoded value

It's becomes very hard to maintain quickly. Maybe it can work, if not touched often.

> Also, it's called expected_true, but... true what? You never actually check for a true value in this test.

"True" here is that there are no exceptions raised.